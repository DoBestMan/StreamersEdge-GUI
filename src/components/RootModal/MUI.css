.root {
  border: 2px solid rgb(190, 195, 210);
  border-radius: 35px;
  background-image: linear-gradient(rgb(0, 22, 104), rgb(6, 0, 78), rgb(6, 0, 42));
}

.register {
  width: 700px;
  border: 2px solid rgb(190, 195, 210);
  border-radius: 35px;
  background-image: linear-gradient(rgb(0, 22, 104), rgb(6, 0, 78), rgb(6, 0, 42));
}

.report {
  width: 700px;
  max-width: 700px;
  height: 700px;
  border: 2px solid rgb(190, 195, 210);
  border-radius: 35px;
  background-image: linear-gradient(to right, rgb(163, 0, 0), rgb(173, 16, 11), rgb(187, 36, 25));
}

.donate {
  width: 35%;
  padding: 30px;
  border: 2px solid rgb(190, 195, 210);
  border-radius: 15px;
  background-image: linear-gradient(rgb(0, 22, 104), rgb(6, 0, 78), rgb(6, 0, 42));
}

.forgot {
  width: 700px;
  border: 2px solid rgb(190, 195, 210);
  border-radius: 35px;
  background-image: linear-gradient(rgb(0, 22, 104), rgb(6, 0, 78), rgb(6, 0, 42));
}
