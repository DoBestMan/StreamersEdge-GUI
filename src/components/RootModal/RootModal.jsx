import React, {Component} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import {withStyles} from '@material-ui/core/styles';
import Dialog from '@material-ui/core/Dialog';
import DialogContent from '@material-ui/core/DialogContent';
import LoginForm from '../Login/LoginForm';
import ForgotPassword from '../ForgotPassword';
import {AppActions, ModalActions, NavigateActions} from '../../actions/';
import {ModalTypes} from '../../constants';
import styles from './MUI.css';

class RootModal extends Component {
  handleClose = () => {
    this.props.setErrorText('');
    this.props.toggleModal();
    this.setState({open: false});
  };

  openPasswordRecovery = () => {
    this.props.setModalType(ModalTypes.FORGOT);
  };

  toggleModalAndRegister = () => {
    this.props.toggleModal();
    this.props.navigateToSignUp();
  };

  render() {
    // Default modal content - a user should NEVER see this in production.
    let modalContent = null;

    // Specify your modals here
    switch (this.props.modalType) {
      case ModalTypes.LOGIN: {
        modalContent = <LoginForm recoverPassword={ this.openPasswordRecovery } goRegister={ this.toggleModalAndRegister } handleLogin={ this.props.login } errorText={ this.props.errorText } />;
        break;
      }

      case ModalTypes.FORGOT: {
        modalContent = <ForgotPassword goRegister={ this.toggleModalAndRegister } />;
        break;
      }

      default: {
        break;
      }
    }

    const {classes} = this.props;
    return (
      <>
        {/* <Dialog open={ this.state.open } onClose={ this.handleClose } aria-labelledby='form-dialog-title' classes={ {paper: classes.dialog-paper__root} }> */}
        <Dialog open={ this.props.isModalOpen } onClose={ this.handleClose } aria-labelledby='form-dialog-title' maxWidth={ 'md' } PaperProps={ {classes: {root: classes.root}} }>
          <DialogContent>{modalContent}</DialogContent>
        </Dialog>
      </>
    );
  }
}

const mapStateToProps = (state) => ({
  isModalOpen: state.getIn(['modal', 'isOpen']),
  modalType: state.getIn(['modal', 'type']),
  errorText: state.getIn(['profiles', 'loginErrorText'])
});

const mapDispatchToProps = (dispatch) => bindActionCreators(
  {
    toggleModal: ModalActions.toggleModal,
    setModalType: ModalActions.setModalType,
    login: AppActions.login,
    setErrorText: AppActions.setLoginError,
    navigateToSignUp: NavigateActions.navigateToSignUp
  },
  dispatch
);

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withStyles(styles)(RootModal));
