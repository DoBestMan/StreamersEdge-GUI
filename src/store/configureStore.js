import {applyMiddleware, createStore} from 'redux';
import {createHashHistory} from 'history';
import {routerMiddleware} from 'connected-react-router';
import Immutable from 'immutable';
import {composeWithDevTools} from 'redux-devtools-extension';
import createRootReducer from '../reducers/index';
import thunk from 'redux-thunk';

// Session history
export const history = createHashHistory();

export default function configureStore() {
  // For typescript, this may need tweaking...
  let initialState = Immutable.Map();

  // Define var for filtering actions when debugging redux to reduce clutter and increase performance.
  let actionsWhitelist = [];

  // Configure enhancer for redux dev tools extensions (if available)
  const composeEnhancers = composeWithDevTools({
    features: {
      dispatch: true
    },
    // Option for immutable
    serialize: {immutable: Immutable},
    actionsWhitelist: actionsWhitelist
  });

  // Construct enhancer
  const enhancer = composeEnhancers(
    applyMiddleware(thunk, routerMiddleware(history))
  );
  const store = createStore(
    createRootReducer(history),
    initialState,
    enhancer
  );

  return store;
}