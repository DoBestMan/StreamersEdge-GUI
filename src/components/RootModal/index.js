import RootModal from './RootModal';

export default RootModal;