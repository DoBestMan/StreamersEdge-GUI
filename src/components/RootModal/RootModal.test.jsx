import * as React from 'react';
import {Provider} from 'react-redux';
import configureStore from '../../store/configureStore';
import {shallow} from 'enzyme';
import RootModal from './RootModal';

describe('<RootModal/>', () => {
  const store = configureStore();
  test('should render correctly', () => {
    const component = shallow(
      <Provider store={ store }>
        <RootModal />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});

describe('<RootModal/>', () => {
  const store = configureStore();
  test('modal should be closed by default', () => {
    const wrapper = shallow(<RootModal store={ store } />);
    const isOpen = wrapper.prop('children').props.isModalOpen;
    expect(isOpen).toBe(false);
  });
});