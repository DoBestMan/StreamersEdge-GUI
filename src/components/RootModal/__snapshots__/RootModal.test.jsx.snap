// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RootModal/> should render correctly 1`] = `ShallowWrapper {}`;
